### ISSUE TITLE
Contribution Ideas/Wishlist

### ISSUE THREAD

**Issue by hendrycks** [pytorch/contrib#12]
Most core functions are already in PyTorch. Here are some which may be better suited for this repository.

### Shake-Shake Regularization
[This regularizer](https://arxiv.org/abs/1705.07485) greatly reduces error rates on CIFAR-10 and CIFAR-100. [Here](https://github.com/hysts/pytorch_shake_shake) is a PyTorch implementation.

### Non-Correlating Multiplicative Noise
[NCMN](https://arxiv.org/abs/1809.07023) is like Shake-Shake regularization but requires far fewer epochs. They have a PyTorch [implementation](https://github.com/zj10/NCMN).

### Random Erasure Data Augmentation
[This](https://arxiv.org/abs/1708.04896) data augmentation technique appears to require less tuning than Cutout and is used in some computer vision papers.
An implementation is [here](https://github.com/zhunzhong07/Random-Erasing).

### GELU
[This](https://openreview.net/pdf?id=Bk0MRI5lg) paper introduced the GELU and Swish (by a different name). This is used in some NLP [work](http://openai-assets.s3.amazonaws.com/research-covers/language-unsupervised/language_understanding_paper.pdf) such as [BERT](https://arxiv.org/pdf/1810.04805.pdf) and NLP [libraries](https://github.com/awslabs/sockeye/pull/229).
```
class GELU(Module):
    def __init__(self, fast=True):
        super(GELU, self).__init__()
        self.fast = fast

    def forward(self, input):
        if self.fast:
            return x * torch.sigmoid(1.702 * x)
        else:
            return 0.5 * x * (1 + torch.tanh(x * 0.7978845608 * (1 + 0.044715 * x * x)))
```

### Swish
[This](https://arxiv.org/pdf/1710.05941v1.pdf) work reinvented x * sigmoid(x) over a year later and greatly popularized the nonlinearity. Later versions add a learnable parameter due to novelty concerns, but its value is less clear. Hence the implementation might look like
```
class Swish(Module):
    def forward(self, input):
        return x * torch.sigmoid(x)  
```

### Convolutional Block Attention Module
[This](http://openaccess.thecvf.com/content_ECCV_2018/html/Sanghyun_Woo_Convolutional_Block_Attention_ECCV_2018_paper.html) iterates on the Squeeze-and-Excitation module by adding a spatial component.
A PyTorch implementation is [here](https://github.com/Jongchan/attention-module). According to the authors, the CBAM usually works better than the BAM.

### To One-Hot
There is no simple, built-in function to convert a list of n labels into a n x num_classes one-hot matrix. 

### SSIM
SSIM is a differentiable loss function used in some computer vision work. A PyTorch implementation is [here](https://github.com/Po-Hsun-Su/pytorch-ssim).

Since this includes several disparate ideas, feel free to close.
